### ISSUE TITLE
Strange Render artifacts on Safari Mobile only

### ISSUE THREAD

**Issue by mix3d** [Kitware/vtk-js#1337]
Throwing this out here in case it might be an implementation specific thing, but on Mobile Safari only I am getting very strange volume render artifacts, as if it is reading the array data in the wrong order. Also reproducible on the Xcode iOS simulator.

A brief summary of my architecture: I am creating a `vtkImageData` object with preallocated `Float32Array` and manually inserting DICOM pixel data  into each slice of the scalar array, then displaying in 3 parallel camera `vtkRenderWindow`s, each with their own `vtkVolume` object (for individual window leveling), but each volume inherits the same imageData object. This works fine in all scenarios save for Mobile Safari.

Since many vtk.js API examples run fine on the iPad, is this just an issue with something I'm doing somehow? Any tips on where to look?

vtk.js version: 12.1.3, but have tried the latest v13 as well, with no difference in behavior.

iOS Safari:
![image](https://user-images.githubusercontent.com/1474137/74899958-895e2800-535b-11ea-9e6e-b972c9dfb966.png)

Literally Any Other computer browser:
![image](https://user-images.githubusercontent.com/1474137/74900025-bc082080-535b-11ea-98cc-84b39f563229.png)

**Comment by jourdain** [comment]
Just to be sure, let me rephrase and restate what I think you did:

You manually created a single vtkImageData by filling its scalars from a DICOM data source.
Then you connected 3 vtkImageMapper to that single vtkImageData and used 3 vtkImageProperty that you added in 3 independent vtkRenderWindow ?

Is that correct? Because the rendering shown at the bottom, those not look like volume rendering...

**Comment by jourdain** [comment]
For what ever reason, it seems that the extent or dimensions of the imageData on the iPad is wrong.
Could this be an invalid DICOM decoding only happening on iPad?

@martinken do you have any clue of what might be happening?

**Comment by mix3d** [comment]
We're using a parallel projection and thin clipping plane to approximate arbitrary volume slicing for MPR, which is why it doesn't look like a volume.

**Comment by jourdain** [comment]
I see, thanks for the clarification.

**Comment by mix3d** [comment]
It was user error after all. My company's Dicom loader apparently grabs a jpeg on mobile devices, I wasn't aware of that behavior. I might try to make the volume work with jpeg data, but forcing it to get the right file is making it work again.

Your recap DID help me pinpoint the problem though! 

Closing the issue.

**Comment by jourdain** [comment]
Glad it help...
